src/plugins/data/server/saved_objects/query.ts

```diff
@@ -11,7 +11,7 @@ import { SavedObjectsType } from 'kibana/server';
 export const querySavedObjectType: SavedObjectsType = {
   name: 'query',
   hidden: false,
-  namespaceType: 'single',
+  namespaceType: 'multiple-isolated',
```

Comment:
The rest of this PR is Step 2 of the Sharing Saved Objects Dev Guide, but this bit is Step 4.
Step 2 should be backported to 7.x, but Step 3 should not be backported, it should only land in 8.0.
 
1. You should split that into a separate PR that is not backported
2. In addition, you can't make this change without also using the `convertToMultiNamespaceTypeVersion` field.